### PR TITLE
Support arm64 resources

### PR DESCRIPTION
**Pull request by ismith**
Circle has these now: https://circleci.com/docs/2.0/arm-resources/?section=executors-and-images

### CODE REVIEW

File: orb.yml [hunk 1]
```diff
@@ -14,24 +14,25 @@ commands:
...
   download_be_executables:
     description: |
       internal buildevents orb command.  don't use this.
     steps:
       - run:
           name: downloading buildevents executables
           command: |
-            curl -q -L -o /tmp/be/bin-linux/buildevents https://github.com/honeycombio/buildevents/releases/download/v0.4.11/buildevents-linux-amd64
-            curl -q -L -o /tmp/be/bin-darwin/buildevents https://github.com/honeycombio/buildevents/releases/download/v0.4.11/buildevents-darwin-amd64
+            curl -q -L -o /tmp/be/bin-linux/buildevents https://github.com/honeycombio/buildevents/releases/download/v0.5.1/buildevents-linux-amd64
```
Comment (lizthegrey) [inline orb.yml:33]: does it make sense to unconditionally download all 3 at the start even if we're throwing away 2 of them?

Comment (ismith) [reply orb.yml:33]: I think it does - because you could have a workflow with multiple machine-types.

Comment (lizthegrey) [reply orb.yml:33]: oh, I see, it downloads once at the start to the shared workspace?

Comment (ismith) [reply orb.yml:33]: That's my understanding, yeah.

**Comment by ismith** [comment]
Notes on testing + for reviewers:

There's a hound branch, https://github.com/honeycombio/hound/pull/6818, that tests this; it runs an `arm64_test` job within a span, and it also manually calls the `buildevents` binary inside it. The passing job at https://github.com/honeycombio/hound/pull/6818/checks?check_run_id=2240628249 lines up with this repo's f8d1a3a.

That commit - f8d1a3a - surprised me a bit; I expected the executable bit set in the previous commit, e0f19a4, to suffice, but it did not. (As evidenced by previous https://github.com/honeycombio/hound/pull/6818 runs failing with `/bin/bash: /tmp/be/bin-linux-arm64/buildevents: Permission denied`.)

**Review by paulosman — APPROVED** [review]
:shipit:

**Pull request merged**